app/controllers/neighborhoods_controller.rb

```diff
@@ -13,14 +15,29 @@ def new
 
   def create
     @neighborhood = Neighborhood.new(neighborhood_params)
+    @neighborhood.user = current_user
     if @neighborhood.save
       redirect_to root_path, notice: "Neighborhood Added!"
     else
-      flash[:alert] = "Your neighborhood was not saved"
       render :new
     end
   end
 
+  def edit
+    @neighborhood = Neighborhood.find(params[:id])
+  end
+
+  def update
+    @neighborhood = Neighborhood.find(params[:id])
+    if @neighborhood.update_attributes(neighborhood_params) &&
+        current_user == @neighborhood.user
```

Comment:
in order to properly authorize the resource here, you should first check before you attempt to update the data. The way this is written, `update_attributes` will run even if the latter half of the conditional returns false